grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/Binary.java

```diff
@@ -18,18 +19,18 @@
  *
  */
 public abstract class Binary extends BinaryBase {
-    private Predicate predicate = null;
+    private IdPredicate predicate = null;
```

Comment:
It was always an Id predicate being injected into a `Binary` Class, so I assumed the change would be safe and I needed the `IdPredicate` so I could get access to the ConceptID without changing the Base Predicate class.